pkg/services/playlist/model.go

```diff
@@ -54,18 +50,18 @@ type Playlists []*Playlist
 //
 
 type UpdatePlaylistCommand struct {
-	OrgId    int64             `json:"-"`
-	UID      string            `json:"uid"`
-	Name     string            `json:"name" binding:"Required"`
-	Interval string            `json:"interval"`
-	Items    []PlaylistItemDTO `json:"items"`
+	OrgId    int64          `json:"-"`
+	UID      string         `json:"uid"`
+	Name     string         `json:"name" binding:"Required"`
+	Interval string         `json:"interval"`
+	Items    []PlaylistItem `json:"items"`
 }
 
 type CreatePlaylistCommand struct {
-	Name     string            `json:"name" binding:"Required"`
-	Interval string            `json:"interval"`
-	Items    []PlaylistItemDTO `json:"items"`
-	OrgId    int64             `json:"-"`
+	Name     string         `json:"name" binding:"Required"`
+	Interval string         `json:"interval"`
+	Items    []PlaylistItem `json:"items"`
```

Comment:
@sakjur -- note that the HTTP API has not changed at all, all the fields (including title+order+etc) are accepted.

The DTO is now only used in the response format, and it drops the unused/unusable fields before sending rather than when we receive them 🤷

Reply:
The response format is part of the HTTP API, not just the request. As unfortunate as the inclusion of these were in the beginning, and as much as I agree with you that they should just be removed, I believe this is a breaking change.

Whether that makes this something that has to wait until Grafana 10 or not is another question. I'd prefer a schedule where that wasn't the case, but our guidance around which breaking changes we tolerate with a note in minors and which have to wait until majors is lacking. If we can mark this as deprecated for 9.3 and remove it in 9.4 that'd be ok as far as I'm concerned (those fields really don't serve any purpose outside of the backend), but I'd prefer to not have to treat every modification to our schemas as a guessworking game of how much we'll mess with someone's integration.

Reply:
I added back unused title -- we can remove it later.

This does remove the internal ids -- but pretty sure that cat is out of the bag with the unannounced breaking change of just using uids in 9

Reply:
That's fair. That was a rather poor decision on our behalf. I supported it, and I regret that. With clear deprecation guidance in place, that would likely have been an introduction of UIDs everywhere in Grafana 9 and a removal of database internal IDs everywhere in Grafana 10. I also think we should move to two major releases per year and a significant downscoping of what we consider minors: there should be no doubt that upgrading to a new minor is a very safe operation, and upgrading to a new major should be something that most do at least after its first minor.